controls/radtreeview/features/ui-virtualization.md

```diff
@@ -71,54 +62,51 @@ Now when you try to expand the "Teams" node, then only those elements that might
 
-#### __XAML__
+#### __[XAML] Example 4: Setting Recycling virtualization mode__
 
 {{region radtreeview-features-ui-virtualization_3}}
-	<telerik:RadTreeView x:Name="radTreeView" telerikTreeView:TreeViewPanel.IsVirtualizing="True" telerikTreeView:TreeViewPanel.VirtualizationMode="Recycling">
-	{{endregion}}
+	<telerik:RadTreeView x:Name="radTreeView" telerikTreeView:TreeViewPanel.IsVirtualizing="True" telerikTreeView:TreeViewPanel.TreeVirtualizationMode="Recycling">
+{{endregion}}
 
-## VirtualizationMode.Hierarchical
+## TreeVirtualizationMode.Hierarchical
 
 When you use __VirtualizationMode__.__Hierarchical__, then items that are out of view will be virtualized. Expanded items are also virtualized. Containers are cached and reused at TreeView level. This mode is suited for indented hierarchies and fully expanded trees. Scrolling may be slower in longer lists (collapsed trees) but faster when the tree is expanded. Expanding and scrolling through a long heavily indented hierarchy should be fast.
 
-#### __XAML__
+#### __[XAML] Example 5: Setting Hierarchical virtualization mode__
 
 {{region radtreeview-features-ui-virtualization_4}}
 	<telerik:RadTreeView x:Name="radTreeView" telerikTreeView:TreeViewPanel.IsVirtualizing="True" telerikTreeView:TreeViewPanel.VirtualizationMode="Hierarchical">
-	{{endregion}}
-
-
+{{endregion}}
 
 > By default, a __TreeViewPanel__ creates an item container for each visible item and discards it when it is no longer needed (such as when the item is scrolled out of view). When an ItemsControl contains a many items, the process of creating and discarding item containers can negatively affect performance. 
 
->When __VirtualizationMode__ is set to __Recycling__, the __TreeViewPanel__ reuses item containers instead of creating a new one each time. When __TreeViewPanel__ cannot recycle item containers, it uses the standard mode of virtualization, which is to create and discard item containers for each item. The following list describes the cases when the __TreeViewPanel__ cannot recycle item containers:
+>When __TreeVirtualizationMode__ is set to __Recycling__, the __TreeViewPanel__ reuses item containers instead of creating a new one each time. When __TreeViewPanel__ cannot recycle item containers, it uses the standard mode of virtualization, which is to create and discard item containers for each item. The following list describes the cases when the __TreeViewPanel__ cannot recycle item containers:
 
 > - The __ItemsControl__ contains item containers of different types. 
 > - You explicitly create the item containers for the __ItemsControl__.
 
-<!-- -->
 >When the __IsVirtualizing__ is set to __True__, the __ChildDefaultLength__ property can be set to the expected header size of the __TreeViewItems__ if it will be different than the default __MinHeight__ of 24 for the __TreeViewItems__.

```

Comment:
We can also consider moving this note to the first place in the article where the IsVirtualizing property is mentioned.